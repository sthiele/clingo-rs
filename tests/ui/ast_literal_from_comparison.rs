use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term1 = ast::Term::from(sym);
    let term2 = ast::Term::from(sym);
    let mut comp = ast::Comparison::gt(term1, term2);
    let lit = ast::Literal::from_comparison(ast::Sign::NoSign, &comp);
    
    comp = ast::Comparison::gt(term1, term2);
    let _end = (lit,comp);
}