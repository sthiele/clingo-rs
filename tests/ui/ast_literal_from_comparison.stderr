error[E0506]: cannot assign to `comp` because it is borrowed
  --> $DIR/ast_literal_from_comparison.rs:10:5
   |
8  |     let lit = ast::Literal::from_comparison(ast::Sign::NoSign, &comp);
   |                                                                ----- borrow of `comp` occurs here
9  |     
10 |     comp = ast::Comparison::gt(term1, term2);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `comp` occurs here
11 |     let _end = (lit,comp);
   |                 --- borrow later used here
